src/frontend/app.ts

```diff
@@ -129,81 +138,95 @@ export default (octokitApp: App): Express => {
 
   app.use(githubClientMiddleware);
 
-  app.use('/api', api);
+  app.use("/", healthcheck);
+
+  app.use("/api", api);
 
   // Add oauth routes
-  app.use('/', oauth.router);
+  app.use("/", oauth.router);
+
+  // Maintenance mode view
+  app.use((req, res, next) => (isMaintenanceMode() ? getMaintenance(req, res) : next()));
+  app.get('/maintenance', csrfProtection, getMaintenance);
 
   app.get(
-    '/github/setup',
+    "/github/setup",
     csrfProtection,
     oauth.checkGithubAuth,
     elapsedTimeMetrics,
-    getGitHubSetup,
+    getGitHubSetup
   );
 
   app.post(
-    '/github/setup',
+    "/github/setup",
     csrfProtection,
     elapsedTimeMetrics,
-    postGitHubSetup,
+    postGitHubSetup
   );
 
   app.get(
-    '/github/configuration',
+    "/github/configuration",
     csrfProtection,
     oauth.checkGithubAuth,
     elapsedTimeMetrics,
-    getGitHubConfiguration,
+    getGitHubConfiguration
   );
 
   app.post(
-    '/github/configuration',
+    "/github/configuration",
     csrfProtection,
     elapsedTimeMetrics,
-    postGitHubConfiguration,
+    postGitHubConfiguration
   );
 
   app.get(
-    '/github/installations',
+    "/github/installations",
     csrfProtection,
     oauth.checkGithubAuth,
     elapsedTimeMetrics,
-    listGitHubInstallations,
+    listGitHubInstallations
   );
 
   app.get(
-    '/github/subscriptions/:installationId',
+    "/github/subscriptions/:installationId",
     csrfProtection,
     elapsedTimeMetrics,
-    getGitHubSubscriptions,
+    getGitHubSubscriptions
   );
 
   app.post(
-    '/github/subscription',
+    "/github/subscription",
     csrfProtection,
     elapsedTimeMetrics,
-    deleteGitHubSubscription,
+    deleteGitHubSubscription
   );
 
   app.get(
-    '/jira/configuration',
+    "/jira/configuration",
     csrfProtection,
     verifyJiraMiddleware,
     elapsedTimeMetrics,
-    getJiraConfiguration,
+    getJiraConfiguration
   );
 
   app.delete(
-    '/jira/configuration',
+    "/jira/configuration",
     verifyJiraMiddleware,
     elapsedTimeMetrics,
-    deleteJiraConfiguration,
+    deleteJiraConfiguration
   );
 
-  app.post('/jira/sync', verifyJiraMiddleware, elapsedTimeMetrics, retrySync);
+  app.post("/jira/sync", verifyJiraMiddleware, elapsedTimeMetrics, retrySync);
+
+
+  // Set up event handlers
+  app.get("/jira/atlassian-connect.json", getJiraConnect);
+  app.post("/jira/events/disabled", jiraAuthenticate, postJiraDisable);
+  app.post("/jira/events/enabled", jiraAuthenticate, postJiraEnable);
+  app.post("/jira/events/installed", postJiraInstall); // we can't authenticate since we don't have the secret
+  app.post("/jira/events/uninstalled", jiraAuthenticate, postJiraUninstall);
```

Comment:
I wonder why did we move it here from `jira.ts` ? 

I find it more confusing than it was before because these are not frontend endpoints and now they are in `frontend/app.ts` file. (We also inlining stuff to the already big method)

Reply:
There's  a large refactor that needs to happen.  The reason for the move is that this app is spaghetti and aren't using the best express practices on top of how probot has it's own kinda server that we're using on top, but just uses the `express.Router`.  It's a mindfield.  

I moved it to frontend to simplify and be able to clearly see at what point maintenance mode stops any following routes.  When they were separate, anything that was inside of frontend couldn't stop routes that were external to it and vice versa.  We needed a way to make it more granular without having to duplicate code everywhere.

There's a much larger task here to how to split up the app correctly using express best practices.  This is the middle ground until that change is done.